File: Solutions/1460- Make Two Arrays Equal by Reversing Subarrays/Solution.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public boolean canBeEqual(int[] target, int[] arr) {
        Map<Integer,Integer> map = new HashMap<>();

        for (int item : arr) {
            if (!map.containsKey(item)) {
                map.put(item, 1);
            } else {
                map.put(item, map.get(item) + 1);
            }
        }

        for (int item : target) {
            if (!map.containsKey(item)) {
                return false;
            } else {
                int temp = map.get(item);
                if (temp > 1) {
                    map.put(item, temp - 1);
                } else {
                    map.remove(item);
                }
            }
        }

        return true;

    }
}

File: Solutions/1460- Make Two Arrays Equal by Reversing Subarrays/Solution.py
class Solution:
    def canBeEqual(self, target: List[int], arr: List[int]) -> bool:
        my_dict = dict()
        for item in arr:
            if item not in my_dict:
                my_dict[item] = 1
            else:
                my_dict[item] += 1
        
        for item in target:
            if item not in my_dict:
                return False
            else:
                if my_dict[item] > 1:
                    my_dict[item] -= 1
                else:
                    my_dict.pop(item)
        
        return True